### PR TITLE
[Merged by Bors] - feat(CompactOpen): add instances

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/CompactOpen.lean [hunk 1]
```diff
@@ -200,14 +200,44 @@ lemma isClopen_setOf_mapsTo (hK : IsCompact K) (hU : IsClopen U) :
...
+  .of_lift'_closure_le fun f ↦ by
+    rw [← tendsto_id', tendsto_nhds_compactOpen]
+    intro K hK U hU hf
+    rcases (hK.image f.continuous).exists_isOpen_closure_subset (hU.mem_nhdsSet.2 hf.image_subset)
+      with ⟨V, hVo, hKV, hVU⟩
+    filter_upwards [mem_lift' (eventually_mapsTo hK hVo (mapsTo'.2 hKV))] with g hg
+    refine ((isClosed_setOf_mapsTo isClosed_closure K).closure_subset ?_).mono_right hVU
+    exact closure_mono (fun _ h ↦ h.mono_right subset_closure) hg
+
+instance [T3Space Y] : T3Space C(X, Y) := inferInstance
```
Comment (eric-wieser) [inline Mathlib/Topology/CompactOpen.lean:240]: Perhaps these should be `example`s if `inferInstance` can find them.

File: Mathlib/Topology/CompactOpen.lean [hunk 2]
```diff
@@ -200,14 +200,44 @@ lemma isClopen_setOf_mapsTo (hK : IsCompact K) (hU : IsClopen U) :
     IsClopen {f : C(X, Y) | MapsTo f K U} :=
   ⟨isClosed_setOf_mapsTo hU.isClosed K, isOpen_setOf_mapsTo hK hU.isOpen⟩
 
+@[norm_cast]
+lemma specializes_coe {f g : C(X, Y)} : (f : X → Y) ⤳ g ↔ f ⤳ g := by
```
Comment (eric-wieser) [inline Mathlib/Topology/CompactOpen.lean:204]: Does
```suggestion
lemma specializes_coe {f g : C(X, Y)} : ⇑f ⤳ ⇑g ↔ f ⤳ g := by
```
work here too?

**Review by eric-wieser — APPROVED** [review]
Are any of these instances docstring worthy?

bors d+

File: Mathlib/Topology/CompactOpen.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Topology/CompactOpen.lean:240]: IMHO, a short-circuit high priority instance makes sense.

**Pull request closed without merging**